cookies/views.py

```diff
@@ -1179,3 +1180,41 @@ def resource_content(request, resource_id):
                 target += '?accessToken=' + auth_token
         return HttpResponseRedirect(target)
     return HttpResponse('Nope')
+
+def export_coauthor_data(request):
+    context = RequestContext(request, {})
+
+    graph = nx.Graph()
+
+    parent_id = request.GET.get('parent_collection', None)
+
+
+    if parent_id:
+        parent_collection = _get_collection_by_id(request, int(parent_id))
+        check_auth = authorization.check_authorization('view_collection', request.user, parent_collection)
+        if not check_auth:
+            return HttpResponse('You do not have permission to view this collection', status=401)
+
+        qset_resources = parent_collection.native_resources.filter(content_resource=False, hidden=False)
+        resources = ResourceFilter(request.GET, queryset=qset_resources)
+
+
+        for resource in resources:
```

Comment:
This is for lines 1202-1212. So, that's a whole lot of code for a pretty simple operation, and a lot of database overhead. 

So you've got database calls happening in three different places: 

1. On 1202 you are loading all columns for all of the selected rows in the Resource table. 
2. On 1204 you're loading all columns for some rows in the Relation table -- this will trigger a separate call for each row that you loaded from Resource. 
3. On 1205, when you access ``relation.predicate.name`` you're spanning other tables, so more hits on the database.

Take a look at [QuerySet.select_related](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#select-related) and [QuerySet.prefetch_related](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#prefetch-related). These will cut down some of the calls if you're spanning relations. 

Then take a look at [QuerySet.values()](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#values) and [QuerySet.values_list()](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#values-list) to get more ideas about how to make the transactions smaller -- if you're only using one or two columns, this can have a huge impact on performance.